test/middleware/routing_middleware_test.dart

```diff
@@ -0,0 +1,423 @@
+        bool handler2Called = false;
+        bool fallbackCalled = false;
+
+        router1.add(Method.get, '/router1/:item',
+            (final RequestContext ctx) async {
+          handler1Called = true;
+          return (ctx as RespondableContext).withResponse(FakeResponse(201));
+        });
+        router2.add(Method.get, '/router2/:data',
+            (final RequestContext ctx) async {
+          handler2Called = true;
+          return (ctx as RespondableContext).withResponse(FakeResponse(202));
+        });
+
+        final pipeline = middleware1
+            .meddle(middleware2.meddle((final RequestContext ctx) async {
+          fallbackCalled = true;
+          return (ctx as RespondableContext).withResponse(FakeResponse(404));
+        }));
+
+        final initialCtx = FakeRequest('/neither/nor').toContext(Object());
+        final resultingCtx = await pipeline(initialCtx);
+        final response =
+            (resultingCtx as ResponseContext).response as FakeResponse;
+
+        expect(handler1Called, isFalse);
+        expect(handler2Called, isFalse);
+        expect(fallbackCalled, isTrue);
+        expect(response.statusCode, equals(404));
+      });
+    });
+
+    group('Nested RoutingMiddleware (via Router.attach)', () {
+      test(
+          'Given a main Router with a nested Router attached, and RoutingMiddleware for the main Router, '
+          'When a request matches a route within the nested Router, '
+          'Then the handler from the nested Router is executed with merged path parameters',
+          () async {
+        Map<Symbol, String>? capturedParams;
+        bool nestedHandlerCalled = false;
+
+        final mainRouter = Router<Handler>();
+        final nestedRouter = Router<Handler>();
+
+        nestedRouter.add(Method.get, '/details/:detailId',
+            (final RequestContext ctx) async {
+          nestedHandlerCalled = true;
+          capturedParams = ctx.pathParameters;
+          return (ctx as RespondableContext)
+              .withResponse(FakeResponse(200, parameters: capturedParams));
+        });
+
+        // Attach nestedRouter to mainRouter under /resource/:resourceId
+        mainRouter.attach('/resource/:resourceId', nestedRouter);
+
+        final mainMiddleware = RoutingMiddleware(mainRouter);
+        final pipeline = mainMiddleware.meddle(
+            (final RequestContext ctx) async =>
+                (ctx as RespondableContext).withResponse(FakeResponse(404)));
+
+        final initialCtx =
+            FakeRequest('/resource/abc/details/xyz').toContext(Object());
+        final resultingCtx = await pipeline(initialCtx);
+        final response =
+            (resultingCtx as ResponseContext).response as FakeResponse;
+
+        expect(nestedHandlerCalled, isTrue);
+        expect(response.statusCode, equals(200));
+        expect(capturedParams, isNotNull);
+        expect(capturedParams, equals({#resourceId: 'abc', #detailId: 'xyz'}));
```

Comment:
Interesting, given that the child route could not possibly know it will be mounted.

But at least the behavior is not specified and fixed.

Reply:
Yes, this speaks to the difference between attaching routers as sub-routers of each other, and nesting middleware.
Since routers currently consume the full path, you have to attach them to nest, but it really should be possible to nest without attaching, and the nested router should only see the _remaining path_, and hence the parameters in the _remaining_ path.

This is something to revisit once #70 lands.